cmd/juju/caas/add.go

```diff
@@ -239,3 +245,17 @@ func addCredentialToLocal(cloudName string, newCredential cloud.Credential, cred
 	}
 	return nil
 }
+
+func (c *AddCAASCommand) addCredentialToController(apiClient CloudAPI, newCredential cloud.Credential, credentialName string) error {
+	currentAccountDetails, err := c.CurrentAccountDetails()
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	cloudCredTagString := fmt.Sprintf("cloudcred-%s_%s_%s", c.caasName, currentAccountDetails.User, credentialName)
```

Comment:
I'm pretty sure you should be using `names.NewCloudCredentialTag` here. Then call `String()` on the resulting tag to get the tag string.